flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/batch/sql/CalcITCase.scala

```diff
@@ -1835,6 +1938,18 @@ class CalcITCase extends BatchTestBase {
     checkResult(
       "SELECT IF(a = '' OR a IS NULL, 'a', 'b') FROM MyTable",
       Seq(row('a'), row('b'), row('a')))
+    checkResult(
+      "SELECT IF(a IN ('', ' ') OR a IS NULL, 'a', 'b') FROM MyTable",
+      Seq(row('a'), row('b'), row('a')))
+    checkResult(
+      "SELECT IF(a IN ('', ' ') OR a IS NOT NULL, 'a', 'b') FROM MyTable",
+      Seq(row('a'), row('a'), row('b')))
+    checkResult(
+      "SELECT IF(a NOT IN ('', ' ') OR a IS NULL, 'a', 'b') FROM MyTable",
+      Seq(row('b'), row('a'), row('a')))
+    checkResult(
+      "SELECT IF(a NOT IN ('', ' ') OR a IS NOT NULL, 'a', 'b') FROM MyTable",
+      Seq(row('a'), row('a'), row('b')))
```

Comment:
Results of the 2nd and the 4th test are the same. You might need to add more test data to make them different.

Reply:
not sure that more data help here...
the issue is that the difference for `IN`/`NOT INT` in these cases is compensated by same ` OR IS NOT NULL`.

Instead I added `nullAs == RexUnknownAs.FALSE` and `nullAs == RexUnknownAs.UNKNOWN` cases